Documentation/Design/Classes/Class Declarations.md

```diff
@@ -0,0 +1,121 @@
+# Class Declarations
+
+## Feature Overview
+
+This improvement adds support for class declarations to GLS, including:
+
+* Naming a class
+* Declaring any generics *(optional)*
+* Declaring a class to extend *(optional)*
+
+
+## Language Examples
```

Comment:
I think it might make sense to put language examples after the GL code. People will care more about that.